phpBB/phpbb/group/helper.php

```diff
@@ -37,4 +91,198 @@ public function get_name($group_name)
+		{
+			if (($mode === 'full' && empty($profile_url)) || $mode === 'no_profile' || $s_is_bots)
+			{
+				$group_name_string = str_replace(array('{GROUP_COLOUR}', '{GROUP_NAME}'), array($group_colour, $group_name), (!$group_colour) ? $this->name_strings['tpl_noprofile'] : $this->name_strings['tpl_noprofile_colour']);
+			}
+			else
+			{
+				$group_name_string = str_replace(array('{PROFILE_URL}', '{GROUP_COLOUR}', '{GROUP_NAME}'), array($profile_url, $group_colour, $group_name), (!$group_colour) ? $this->name_strings['tpl_profile'] : $this->name_strings['tpl_profile_colour']);
+			}
+		}
+
+		$name_strings = $this->name_strings;
+
+		/**
+		 * Use this event to change the output of the group name
+		 *
+		 * @event core.modify_group_name_string
+		 * @var string	mode				profile|group_name|colour|full|no_profile
+		 * @var int		group_id			The group identifier
+		 * @var string	group_name			The group name
+		 * @var string	group_colour		The group colour
+		 * @var string	custom_profile_url	Optional parameter to specify a profile url.
+		 * @var string	group_name_string	The string that has been generated
+		 * @var array	name_strings		Array of original return templates
+		 * @since 3.2.6-RC1
+		 */
+		$vars = array(
+			'mode',
+			'group_id',
+			'group_name',
+			'group_colour',
+			'custom_profile_url',
+			'group_name_string',
+			'name_strings',
+		);
+		extract($this->dispatcher->trigger_event('core.modify_group_name_string', compact($vars)));
+
+		return $group_name_string;
+	}
+
+	/**
+	 * Get group rank title and image
+	 *
+	 * @param array		$group_data		the current stored group data
+	 *
+	 * @return array					An associative array containing the rank title (title),
+	 * 									the rank image as full img tag (img) and the rank image source (img_src)
+	 */
+	public function get_rank($group_data)
+	{
+		$group_rank_data = array(
+			'title'		=> null,
+			'img'		=> null,
+			'img_src'	=> null,
+		);
+
+		/**
+		 * Preparing a group's rank before displaying
+		 *
+		 * @event core.get_group_rank_before
+		 * @var	array	group_data		Array with group's data
+		 * @since 3.2.6-RC1
+		 */
+
+		$vars = array('group_data');
+		extract($this->dispatcher->trigger_event('core.get_group_rank_before', compact($vars)));
+
+		if (!empty($group_data['group_rank']))
+		{
+			// Only obtain ranks if group rank is set
+			$ranks = $this->cache->obtain_ranks();
+
+			if (isset($ranks['special'][$group_data['group_rank']]))
+			{
+				$rank = $ranks['special'][$group_data['group_rank']];
+
+				$group_rank_data['title'] = $rank['rank_title'];
+
+				$group_rank_data['img_src'] = (!empty($rank['rank_image'])) ? $this->path_helper->update_web_root_path($this->phpbb_root_path . $this->config['ranks_path'] . '/' . $rank['rank_image']) : '';
+
+				$group_rank_data['img'] = (!empty($rank['rank_image'])) ? '<img src="' . $group_rank_data['img_src'] . '" alt="' . $rank['rank_title'] . '" title="' . $rank['rank_title'] . '" />' : '';
```

Comment:
@marc1706 I would be more in favor of factoring this type out as it prevents us from using svgs

Reply:
This is html as well

Reply:
Done